kubermatic/provider.go

```diff
@@ -100,45 +102,65 @@ func Provider() *schema.Provider {
 }
 
 func configure(d *schema.ResourceData, terraformVersion string, fd *os.File) (interface{}, error) {
+	var logdev, logdebug bool
+	var logpath, host, token, tokenPath string
+	if v, ok := d.Get("development").(bool); ok {
+		logdev = v
+	}
+	if v, ok := d.Get("debug").(bool); ok {
```

Comment:
you're checking if type assertion is ok here and assign value only when it's `true`, in case we change parameter type in schema this not gonna throw any error, it will skip the parameter. That's why I let the panic to be raised in the past, I would suggest to throw an error or let type assertion to raise a panic.

Reply:
I think some options are optional right? Like `debug`, `development`, `tokenPath`. If so, for ex if `debug` is not specified, `d.Get("debug")` throws an error, because d.Get return nil.

https://play.golang.org/p/mdoFM8I1VVv

Reply:
in that case `d.Get` would not throw an error because these options are defined in schema
```
// Get returns the data for the given key, or nil if the key doesn't exist
// in the schema.
//
// If the key does exist in the schema but doesn't exist in the configuration,
// then the default value for that type will be returned. For strings, this is
// "", for numbers it is 0, etc.
```

and if you want to check if something is set you should use `GetOk`
```
// If you want to test if something is set at all in the configuration,
// use GetOk.
```

so in that we don't have to check if type assertion is ok

https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_data.go#L75

Reply:
oh I see, it is never nil.